dashboard/data/bmg_plate.py

```diff
@@ -149,6 +149,53 @@ def well_to_ids(well_name: str) -> tuple[int, int]:
     return ord(head) - 65, int(tail) - 1
 
 
+def parse_bmg_file_iostring(filename: str, filecontent: str) -> np.array:
+    """
+    Read data from iostring file to np.array
+
+    :return: array with plate values
+    """
+    plate = np.zeros(shape=(16, 24))
+    barcode = filename.split(".")[0].split("\\")[-1]
+    content = []
+    line = filecontent.readline()
+    while line:
+        content.append(line)
+        line = filecontent.readline()
+    for i, line in enumerate(content):
+        # handle different formatting (as in FIREFLY)
+        if i == 0 and "A" not in line:
+            df = pd.read_csv(content, header=None)
+            plate = df.to_numpy()
+            break
+        well, value = line.split()
+        i, j = well_to_ids(well)
+        plate[i, j] = value
+    return barcode, plate
+
+
+def parse_bmg_files_from_iostring(
+    files: tuple[(str, str)]
+) -> tuple[pd.DataFrame, np.ndarray]:
+    """
+    Parse file from iostring with BMG files to DataFrame
+
+    :param files: tuple containing names and content of files
+    :return: DataFrame with BMG files (=plates) as rows
+    """
+    plate_summaries = []
+    plate_values = []
+    for filename, filecontent in files:
+        barcode, plate_array = parse_bmg_file_iostring(filename, filecontent)
+        plate = Plate(barcode, plate_array)
+        z_wo, outliers_mask = calculate_z_outliers(plate)
+        plate_summaries.append(get_summary_tuple(plate, z_wo))
+        plate_values.append([plate.plate_array, outliers_mask])
+    df = pd.DataFrame(plate_summaries)
+    plate_values = np.asarray(plate_values)
+    return df, plate_values
+
+
 def parse_bmg_file(filepath: str) -> tuple[str, np.ndarray]:
```

Comment:
would it be possible to make it work without having to create a file? ideally we should handle all this in memory so no file reading - iostrings should suffice.

In general it is good to separate IO operations (such as reading from a file) from processing the data